src/main/java/uk/ac/ebi/eva/submission/controller/admin/AdminController.java

```diff
@@ -56,6 +57,21 @@ public ResponseEntity<?> markSubmissionStatus(@PathVariable("submissionId") Stri
         }
     }
 
+    @Operation(summary = "This endpoint retrieves all the submissions from the database with given step and status")
+    @Parameters({
+            @Parameter(name="step", description = "The processing step of the submission.",
+                    required = true, in= ParameterIn.PATH),
+            @Parameter(name="status", description = "The status of the submission processing step.",
+                    required = true, in= ParameterIn.PATH)
+    })
+
+    @GetMapping("submission/{submissionId}")
+    public ResponseEntity<?> getSubmissionsDetail(
+            @PathVariable("submissionId") String submissionId) {
+        SubmissionDetails submissionDetail = submissionService.getSubmissionDetail(submissionId);
+        return new ResponseEntity<>(submissionDetail, HttpStatus.OK);
```

Comment:
Should this also handle `SubmissionDoesNotExistException` like the other endpoints?

Reply:
Done